icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -642,9 +613,10 @@ SimpleDateFormat& SimpleDateFormat::operator=(const SimpleDateFormat& other)
     }
 
     UErrorCode localStatus = U_ZERO_ERROR;
-    freeFastNumberFormatters();
-    initFastNumberFormatters(localStatus);
-
+    // SimpleNumberFormatter does not have a copy constructor. Furthermore,
+    // it references data from an internal field, fNumberFormatter,
+    // so we must rematerialize that reference after copying over the number formatter
+    initSimpleNumberFormatter(fLocale, localStatus);
```

Comment:
FYI -- note to self:
It looks like construct() and initialize() and initSimpleNumberFormatter() take a Locale parameter but it's always fLocale. In a follow-up PR we should try to remove that and just use fLocale in those functions.